### PR TITLE
fix: add getDeviceType and getDeviceTypeSync mocks

### PR DESCRIPTION
**Pull request by maximilize**
## Description

The mocks `getDeviceType` and `getDeviceTypeSync` were missing after updating to v10.

### CODE REVIEW

**Review by schie — APPROVED** [review]
LGTM — Thanks!

File: jest/react-native-device-info-mock.js [hunk 1]
```diff
@@ -37,6 +37,8 @@ const diMock = {
   getDeviceName: stringFnAsync(),
   getDeviceNameSync: stringFnSync(),
   getDeviceToken: stringFnAsync(),
+  getDeviceType: stringFnAsync(),
```
Comment (mikehardy) [inline jest/react-native-device-info-mock.js:40]: This duplicates the existing one, but the existing `getDeviceType` needs to change to `stringFnAsync()`

**Review by mikehardy — CHANGES_REQUESTED** [review]
Thanks for posting a PR to sync up the mocks, much appreciated - looks like it needs a small tweak?

File: jest/react-native-device-info-mock.js (hunk 1 above)
Comment (schie) [reply jest/react-native-device-info-mock.js:40]: gah! this is what I get for reviewing before the tests complete!

File: jest/react-native-device-info-mock.js [hunk 2]
```diff
@@ -122,6 +122,7 @@ const diMock = {
   getBundleId: stringFnSync(),
   getDeviceId: stringFnSync(),
   getDeviceType: stringFnSync(),
```
Comment (mikehardy) [inline jest/react-native-device-info-mock.js:124]: This one needs to be stringFnAsync now I think?
```suggestion
  getDeviceType: stringFnASync(),
```

**Review by mikehardy — CHANGES_REQUESTED** [review]
new signature is in and looks good, but old needs to be async now I think?

no good deed unpunished! :sweat_smile:

File: jest/react-native-device-info-mock.js (hunk 2 above)
Comment (maximilize) [reply jest/react-native-device-info-mock.js:124]: The original function is also not async, so I guess it's fine like this.

See https://github.com/react-native-device-info/react-native-device-info/blob/master/src/index.ts#L565

Comment (mikehardy) [reply jest/react-native-device-info-mock.js:124]: Sorry for the delay! You are correct

**Review by mikehardy — APPROVED** [review]

**Pull request merged**